pkg/logs/input/tailer/file_provider.go

```diff
@@ -0,0 +1,78 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2017 Datadog, Inc.
+
+package tailer
+
+import (
+	"log"
+	"path/filepath"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/logs/config"
+)
+
+// File represents a file we want to tail
+type File struct {
+	Path   string
+	Source *config.IntegrationConfigLogSource
+}
+
+// NewFile returns a new File
+func NewFile(path string, source *config.IntegrationConfigLogSource) *File {
+	return &File{
+		Path:   path,
+		Source: source,
+	}
+}
+
+// FileProvider implements the logic to retrieve at most filesLimit Files defined in sources
+type FileProvider struct {
+	sources    []*config.IntegrationConfigLogSource
+	filesLimit int
+}
+
+// NewFileProvider returns a new FileProvider
+func NewFileProvider(sources []*config.IntegrationConfigLogSource, filesLimit int) *FileProvider {
+	return &FileProvider{
+		sources:    sources,
+		filesLimit: filesLimit,
+	}
+}
+
+// NewFiles returns all the Files matching paths in sources,
+// it cannot return more than filesLimit Files
+// For now, there is no way to prioritize specific Files over others
+func (r *FileProvider) NewFiles() []*File {
```

Comment:
Good comment. One suggestion:
One problem we might encounter is that if we reach the file limit this could return different files at each iteration. I think we're ready to accept the consequences, but could we comment it heavily?

Reply:
Agreed